pkg/resource/predicates.go

```diff
@@ -41,20 +41,33 @@ func NewPredicates(fn PredicateFn) predicate.Funcs {
 	}
 }
 
-// TODO(negz): Use separate predicates for resource claims and managed
-// resources once controller-runtime supports this.
-// https://github.com/kubernetes-sigs/controller-runtime/issues/572
-
-// HasClassReferenceKinds accepts objects that reference the supplied
-// non-portable class kind, either directly or indirectly via a portable class
-// kind.
-func HasClassReferenceKinds(c client.Client, oc runtime.ObjectCreater, k ClassKinds) PredicateFn {
+// AnyOf accepts objects that pass any of the supplied the supplied predicate
+// functions.
+func AnyOf(fn ...PredicateFn) PredicateFn {
 	return func(obj runtime.Object) bool {
-		if _, ok := obj.(NonPortableClassReferencer); ok {
-			return HasDirectClassReferenceKind(NonPortableClassKind(k.NonPortable))(obj)
+		for _, f := range fn {
+			if f(obj) {
+				return true
+			}
+		}
+		return false
+	}
+}
+
+// HasManagedResourceReferenceKind accepts objects that reference the supplied
+// managed resource kind.
+func HasManagedResourceReferenceKind(k ManagedKind) PredicateFn {
+	return func(obj runtime.Object) bool {
+		r, ok := obj.(ManagedResourceReferencer)
+		if !ok {
+			return false
 		}
 
-		return HasIndirectClassReferenceKind(c, oc, k)(obj)
+		if r.GetResourceReference() == nil {
+			return false
+		}
+
+		return r.GetResourceReference().GroupVersionKind() == schema.GroupVersionKind(k)
```

Comment:
i was wondering about this the other day, do we need to do any `ToLower()` normalization or `EqualFold()` comparison in these predicates?

Reply:
The type metadata supplied to the predicate will always match the actual case of our API types (e.g. `RDSInstanceClass`, not `rdsinstanceclass`), so the question is whether we want to require our users to match the actual case. I'm ambivalent about that so I'll avoid changing it for the time being. Everything will work fine as long as the correct case is used. It seems typical to do this in Kubernetes, i.e. I usually see `kind: Deployment`, not `kind: deployment`.